File: bee_3151.rb
H, Z, L = gets.chomp.split.map(&:to_i)

if (H < Z && Z < L) || (L < Z && Z < H)
    puts "zezinho"
elsif (Z < H && H < L) || (L < H && H < Z)
    puts "huguinho"
else
    puts "luisinho"
end

File: bee_3151.c

#include <stdio.h>

int main() {
    int H, Z, L;
    scanf("%d %d %d", &H, &Z, &L);

    if ((H < Z && Z < L) || (L < Z && Z < H)) {
        printf("zezinho\n");
    } else if ((Z < H && H < L) || (L < H && H < Z)) {
        printf("huguinho\n");
    } else {
        printf("luisinho\n");
    }

    return 0;
}